File: QuickStart/ChartView/FirstLook/Legend.vb
Imports System
Imports Telerik.WinControls
Imports Telerik.WinControls.UI
Imports System.Drawing

Namespace Telerik.Examples.WinControls.ChartView.FirstLook
	Public Class Legend
		Inherits RadControl
		Private legendElement As LegendElement

		Public Sub New()
		End Sub

		Protected Overrides Sub CreateChildItems(ByVal parent As RadElement)
			MyBase.CreateChildItems(parent)
			legendElement = New LegendElement()
			parent.Children.Add(legendElement)

			Me.BackColor = Color.White
		End Sub

		Public Sub BuildLegend(ByVal element As RadChartElement)
			Me.legendElement.BuildLegend(element)
		End Sub
	End Class
End Namespace

File: QuickStart/ChartView/FirstLook/Legend.cs
using System;
using Telerik.WinControls;
using Telerik.WinControls.UI;
using System.Drawing;

namespace Telerik.Examples.WinControls.ChartView.FirstLook
{
    public class Legend: RadControl
    {
        LegendElement legendElement;

        public Legend()
        { }

        protected override void CreateChildItems(RadElement parent)
        {
            base.CreateChildItems(parent);
            legendElement = new LegendElement();
            parent.Children.Add(legendElement);

            this.BackColor = Color.White;
        }

        public void BuildLegend(RadChartElement element)
        {
            this.legendElement.BuildLegend(element);
        }
    }
}
